src/datachain/catalog/catalog.py

```diff
@@ -509,6 +509,60 @@ def clone_catalog_with_cache(catalog: "Catalog", cache: "Cache") -> "Catalog":
     return clone
 
 
+def get_all_ids(d: dict[str, Any]) -> list[int]:
+    """
+    Extracts all unique dataset IDs from a nested dictionary structure.
+    """
+    ids = set()
+    for key, value in d.items():
+        ids.add(int(key))
+        if isinstance(value, dict):
+            ids.update(get_all_ids(value))
+    return sorted(ids)
+
+
+def build_nested_dependencies(
```

Comment:
can we also check if there are some libs that build graph from a connections matrix (that what is happening here, right?)

Reply:
I am not sure what you mean by connections matrix tbh. But why do we want to introduce a dependency to do something this simple? 
I mean we are just recursively attaching the dependencies based on the tree we already have. For example if we have: 

```
{"306": {"297": {"255": {"250": {}}, "256": {"252": {}}}}}
```

We will get the dependency we have cached in the variable to form a tree:
`<Dependency (306)> .dependencies = [<Dependency(297)...`. and so on.

This is verified and tested in test_dependency.